### ISSUE TITLE
Add support for user defined parameters in `ConditionContext`

### ISSUE THREAD

**Issue by ghardin1314** [nucypher/nucypher-ts#173]
I think we discussed some potential updates to the `ConditionContext` api (and concept in general) previously. One thing the should be added is support for user defined parameter inputs. 

`Condition`(s) allow for user supplied parameters via a `:<parameter_name>` (i.e. `:tokenId`) but there is currently no way to supply these at decryption time as the context returns only `:userAddress` when making a request to porter here: 

https://github.com/nucypher/nucypher-ts/blob/3c4d2f6186bee3e036e9f8c4d6f713e0d94a9857/src/policies/conditions.ts#L461-L471

Example of support from nucypher core library

https://github.com/nucypher/nucypher/blob/development/tests/acceptance/blockchain/conditions/test_conditions.py#L213

This is my current hack to make it work on my test project

```typescript
class MasterfileConditionConditionContext extends ConditionContext {
  private superToJson = this.toJson;

  constructor(
    conditionSet: ConditionSet,
    web3Provider: Web3Provider,
    private readonly customParams: Record<string, any>
  ) {
    super(conditionSet, web3Provider as any);
  }

  toJson = async (): Promise<string> => {
    const params = JSON.parse(await this.superToJson());
    return JSON.stringify({ ...this.customParams, ...params });
  };
}
```

**Comment by piotr-roslaniec** [comment]
Related issue: #138

**Comment by derekpierre** [comment]
@piotr-roslaniec - #138  is specific to the returnValueTest, but can user-defined context variables not be included in the `ConditionContext` object in some fashion, and included in the json?  For example, what about specifying user-context variables as parameters?

**Comment by derekpierre** [comment]
@ghardin1314 is this issue currently "workaroundable" for you? Just trying to organize releases.

**Comment by ghardin1314** [comment]
@derekpierre yes we have a work around for this right now

**Comment by piotr-roslaniec** [comment]
Closed by #199

**Comment by derekpierre** [comment]
This is partially closed by #199 which has some outstanding bugs. The follow-up issue to fix outstanding bugs is #205.
